portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -14,35 +14,43 @@
 
 package com.google.sps.servlets;
 
-import com.google.appengine.api.users.UserService;
-import com.google.appengine.api.users.UserServiceFactory;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet("/")
-public class HomeServlet extends HttpServlet {
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
+/**
+ * Servlet responsible for authenticating a user before they can comment
+ */
+@WebServlet("/auth-check")
+public class AuthenticationServlet extends DataServlet {
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    List<String> loginStatus = new ArrayList<>();
```

Comment:
I would prefer to have this use a class rather than a List/Array of strings.
Makes it clearer what the components are instead of needing to make sure the array indices are always the same. Also makes it more robust in case you decide that more information needs to be sent.

So, something like a perhaps inner class:
```
  class LoginStatus {
      private bool loggedIn;
      private String loginURL;
      private String logout URL;
      LoginStatus (bool loggedIn, String loginURL, String logoutURL) {...}
 }

// You can decide if you want to create both types of URL or just one and leave the other empty.
LoginStatus loginStatus = new LoginStatus(userService.isUserLoggedIn(), ...);
Gson gson new Gson();
response.getWriter().println(gson.toJson(loginStatus));
```

This will require changing the Gson call's argument type in your DataServlet function.
Although these are all so simple, that repeating the code would be acceptable.

https://sites.google.com/site/gson/gson-user-guide#TOC-Object-Examples (Assuming you can get there)

Reply:
That's a really good suggestion, Tom! Thanks! I just updated the pull request with this change.